### PR TITLE
[isoltest] Introduce shorthand declaration: make `->` optional.

### PR DESCRIPTION
**Pull request by erak**
Part of https://github.com/ethereum/solidity/issues/6669.

### CODE REVIEW

File: test/libsolidity/util/SoltestTypes.h [hunk 1]
```diff
@@ -249,6 +249,9 @@ struct FunctionCall
 	DisplayMode displayMode = DisplayMode::SingleLine;
 	/// Marks this function call as the constructor.
 	bool isConstructor = false;
+	/// Marks this function call as short-handed, meaning
+	/// no `->` declared.
+	bool isShorthand = true;
```
Comment (chriseth) [inline test/libsolidity/util/SoltestTypes.h]: maybe `omitsArrow`?

Comment (erak) [reply test/libsolidity/util/SoltestTypes.h]: Sounds good!

**Comment by chriseth** [comment]
Needs to be rebased.

**Comment by erak** [comment]
Rebased.

File: test/libsolidity/util/TestFileParser.cpp [hunk 2]
```diff
@@ -95,11 +95,19 @@ vector<dev::solidity::test::FunctionCall> TestFileParser::parseFunctionCalls(siz
...
+					{
+						call.omitsArrow = false;
+						call.expectations = parseFunctionCallExpectations();
+						if (accept(Token::Newline, true))
+							m_lineNumber++;
+					}
+					else
+					{
+						call.expectations.failure = false;
+						call.displayMode = FunctionCall::DisplayMode::SingleLine;
```
Comment (leonardoalt) [inline test/libsolidity/util/TestFileParser.cpp:108]: Why is this needed? Could the arguments be given in a different line?

File: test/libsolidity/util/TestFunctionCall.cpp [hunk 3]
```diff
@@ -74,7 +74,16 @@ string TestFunctionCall::format(
...
 				stream << ws << comment << m_call.arguments.comment << comment;
-			stream << ws << arrow << ws;
+
+			if (m_call.omitsArrow)
+			{
+				if (_renderResult && (m_failure || !matchesExpectation()))
+					stream << ws << arrow;
+			}
+			else
+			{
```
Comment (leonardoalt) [inline test/libsolidity/util/TestFunctionCall.cpp]: Can be removed

**Comment by leonardoalt** [comment]
What happens if a test checks return values but there's no arrow, parsing error?

File: test/libsolidity/util/TestFileParser.cpp (hunk 2 above)
Comment (erak) [reply test/libsolidity/util/TestFileParser.cpp:108]: Right, need to check that particular case.

**Comment by erak** [comment]
@leonardoalt The parser will throw an error for a construct like `// h() FAILURE` (which is also covered by the unit tests https://github.com/ethereum/solidity/pull/7094/files#diff-5be16c388747ac6c28ef428f1c6e7de9R799)

File: test/libsolidity/util/TestFileParser.cpp (hunk 2 above)
Comment (erak) [reply test/libsolidity/util/TestFileParser.cpp:108]: Oh, reading this again, I might have misunderstood your comment. Yes, you can define a test like below. This will set the display mode to multi-line, which will have a different formatting then single-line:
```
contract C {
        function f(uint a, uint b, uint c, uint d, uint e) public returns (uint) {
            return a + b + c + d + e;
        }
    }
// ----
// f(uint256,uint256,uint256,uint256,uint256):
// 1,
// 1,
// 1,
// 1,
// 1
```
The above would be still single-line, since the "newlined" arguments are not affecting the display mode. Only having the arrow on a newline would change the mode. Since this is now the shorthand version without an arrow, the default should be single-line again:
```
Obtained result:
  // f(uint256,uint256,uint256,uint256,uint256):
  // 1,
  // 1,
  // 1,
  // 1,
  // 1 -> 5
```

**Comment by erak** [comment]
Updated.

File: test/libsolidity/util/TestFileParser.cpp (hunk 2 above)
Comment (leonardoalt) [reply test/libsolidity/util/TestFileParser.cpp:108]: Ok!

File: test/libsolidity/util/TestFileParser.cpp [hunk 4]
```diff
@@ -86,20 +86,28 @@ vector<dev::solidity::test::FunctionCall> TestFileParser::parseFunctionCalls(siz
 						call.displayMode = FunctionCall::DisplayMode::MultiLine;
 						m_lineNumber++;
 					}
-
+
```
Comment (Marenz) [inline test/libsolidity/util/TestFileParser.cpp]: please strip your spaces :)

**Review by Marenz — CHANGES_REQUESTED** [review]
You fixed the whitespace and the indent in the the wrong commit.. the first one introduces it, the second one fixes it again. Please add those fixes to the one that introduces them :)

Otherwise LGTM.

**Comment by chriseth** [comment]
No reason to delay this for more than two days.
